homeassistant/components/switch/fritzdect.py

```diff
@@ -120,17 +123,52 @@ def is_on(self):
 
     def turn_on(self, **kwargs):
         """Turn the switch on."""
-        actor = self.data.fritz.get_actor_by_ain(self.data.ain)
-        actor.switch_on()
+        if not self.data.is_online:
+            _LOGGER.error("turn_on: Not online skipping request")
+            return
+
+        try:
+            actor = self.data.fritz.get_actor_by_ain(self.data.ain)
+            actor.switch_on()
+        except (RequestException, HTTPError):
+            _LOGGER.error("Fritz!Box query failed, triggering relogin")
+            self.data.is_online = False
 
     def turn_off(self):
         """Turn the switch off."""
-        actor = self.data.fritz.get_actor_by_ain(self.data.ain)
-        actor.switch_off()
+        if not self.data.is_online:
+            _LOGGER.error("turn_off: Not online skipping request")
+            return
+
+        try:
+            actor = self.data.fritz.get_actor_by_ain(self.data.ain)
+            actor.switch_off()
+        except (RequestException, HTTPError):
+            _LOGGER.error("Fritz!Box query failed, triggering relogin")
+            self.data.is_online = False
 
     def update(self):
         """Get the latest data from the fritz box and updates the states."""
-        self.data.update()
+        if not self.data.is_online:
+            _LOGGER.error("update: Not online, logging back in")
+
+            # Exception is thrown and pylint will complain - ignoring this
+            # pylint: disable=W0703
```

Comment:
This comment that disables broad except warning should be inlined on line 159 after the code. Reason is that we don't want to disable it in other places.

Also please replace W0703 with the text `broad-except`.